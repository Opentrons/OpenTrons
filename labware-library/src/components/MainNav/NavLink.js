// @flow
import * as React from 'react'
import cx from 'classnames'
import type { Submenu } from './types'

import styles from './MainNav.module.css'

export type Props = {|
  ...Submenu,
  className?: string,
  onToggle?: (name: string | null) => void,
  homeUrl: string,
  gtm?: {
    action: string,
    category: string,
    label: string,
  },
  linkout?: boolean,
|}

export function NavLink(props: Props): React.Node {
  const { gtm, className, url, homeUrl } = props

  return (
    <>
      <a
        href={props.linkout ? url : `${homeUrl}${url || ''}`}
        className={cx('link_title', className)}
        target="_blank"
        rel="noopener noreferrer"
        data-gtm-category={`l-${gtm ? gtm.category : ''}`}
        data-gtm-label={gtm ? gtm.label : ''}
        data-gtm-action={gtm ? gtm.action : ''}
      >
        {props.name}
        {props.description && (
          <div className={styles.nav_sub_description}>{props.description}</div>
        )}
      </a>
    </>
  )
}
