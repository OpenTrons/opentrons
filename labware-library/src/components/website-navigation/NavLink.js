// @flow
import * as React from 'react'
import cx from 'classnames'
import styles from './styles.css'

import type { Link } from './types'

export default function NavLink(props: Link) {
  return (
    <div className={styles.link_group}>
      <a
        href={props.url}
        className={cx(styles.link_title, { [styles.link_cta]: props.cta })}
        target="_blank"
        rel="noopener noreferrer"
      >
        {props.name}
        {props.cta && <span>&nbsp; &gt;</span>}
      </a>
      {props.description && (
        <div className={styles.link_description}>{props.description}</div>
      )}
    </div>
  )
}

export function NavButton(props: Link) {
  return (
    <a
      href={props.url}
      className={styles.link_button}
      target="_blank"
      rel="noopener noreferrer"
    >
      {props.name} &nbsp; &gt;
    </a>
  )
}
