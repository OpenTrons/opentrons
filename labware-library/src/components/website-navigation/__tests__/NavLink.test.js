// @flow
// tests for Logo image component
import * as React from 'react'
import { shallow } from 'enzyme'

import NavLink from '../NavLink'

describe('NavLink', () => {
  test('component renders', () => {
    const tree = shallow(<NavLink name="foo" url="bar" />)

    expect(tree).toMatchSnapshot()
  })

  test('component renders name and optional description from props', () => {
    const props = {
      name: 'some link',
      url: '#',
      description: 'some link description',
    }
    const tree = shallow(<NavLink {...props} />)
    expect(tree.find('a').text()).toEqual(props.name)
    expect(tree.find('.link_description').text()).toEqual(props.description)
  })

  test('component renders conditional cta class', () => {
    const tree = shallow(<NavLink name="foo" cta={true} />)
    expect(tree.find('a').hasClass('link_cta')).toEqual(true)
  })
})
