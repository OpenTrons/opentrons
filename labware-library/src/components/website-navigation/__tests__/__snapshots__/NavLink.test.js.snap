// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink component renders 1`] = `
<div
  className="link_group"
>
  <a
    className="link_title"
    href="bar"
    rel="noopener noreferrer"
    target="_blank"
  >
    foo
  </a>
</div>
`;
